src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+            if (!tmpFusedList.isEmpty()) {
+                Iterator<IAtomContainer> tmpFusedIterator = tmpFusedList.iterator();
+                IAtomContainerSet tmpFusedSet = new AtomContainerSet();
+                while (tmpFusedIterator.hasNext()) {
+                    tmpFusedSet.addAtomContainer(tmpFusedIterator.next());
+                }
+                for (IAtomContainer tmpContainer: tmpFusedSet.atomContainers()) {
+                    for (IAtom tmpFusedAtom: tmpContainer.atoms()) {
+                        int tmpAtomInteger = tmpFusedAtom.getProperty(tmpAlkylSFAtomIndexProperty);
+                        tmpAtomArray[tmpAtomInteger].setFlag(CDKConstants.ISINRING,true);
+                        tmpAtomArray[tmpAtomInteger].setFlag(CDKConstants.ISPLACED,false);
+                    }
+                    for (IBond tmpFusedBond: tmpContainer.bonds()) {
+                        int tmpBondInteger = tmpFusedBond.getProperty(tmpAlkylSFBondIndexProperty);
+                        tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISINRING,true);
+                        tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISPLACED, false);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        //</editor-fold>
+        //
+        //<editor-fold desc="ConjugatedPiSystemsDetector" defaultstate="collapsed">
+        try {
+            IAtomContainerSet tmpConjugatedAtomContainerSet;
+            tmpConjugatedAtomContainerSet = ConjugatedPiSystemsDetector.detect(aMolecule);
+            //molecule mapping
+            //iterate over every atomcontainer from ConjPiSystemsDetector output
+            for (IAtomContainer tmpConjAtomContainer: tmpConjugatedAtomContainerSet.atomContainers()) {
+                for (IAtom tmpConjAtom: tmpConjAtomContainer.atoms()) {
+                    int tmpAtomInteger = tmpConjAtom.getProperty(tmpAlkylSFAtomIndexProperty);
+                    tmpAtomArray[tmpAtomInteger].setFlag(CDKConstants.ISCONJUGATED,true);
+                    tmpAtomArray[tmpAtomInteger].setFlag(CDKConstants.ISPLACED,false);
+                }
+                for (IBond tmpConjBond: tmpConjAtomContainer.bonds()) {
+                    int tmpBondInteger = tmpConjBond.getProperty(tmpAlkylSFBondIndexProperty);
+                    tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISCONJUGATED,true);
+                    tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISPLACED,false);
+                }
+            }
+        } catch (Exception anException) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING,
+                    anException + " Molecule ID: " + aMolecule.getID());
+            throw new RuntimeException(anException);
+        }
+        //</editor-fold>
+        //
+        //<editor-fold desc="CycleFinder (Single Rings)" defaultstate="collapsed">
+        CycleFinder tmpMCBCycleFinder = Cycles.mcb();
```

Comment:
I know we already discussed that but sorry, I have to ask again: why not simply use the .isolatedRingFragments() method of RingSearch here? Or maybe only use MCB? Since you later separate the rings and ring systems by whether they are connected or not anyway, right?

Reply:
RingSearch also has the method .ringFragments() to extract isolated and fused ring systems together at once.